### PR TITLE
Add configurable Mech request price parameter and migrate mech contract functionality

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

* Adds parameter 'request_price' 
    * If the optional parameter 'request_price' is set, then use that price (wei) for sending the Mech requests. Otherwise, determine the request price by calling the contract.
    * This parameter is useful to set 'request_price=0' when the agent is using a nevermined subscription.
    * (These are the changes from PR #144 , but has been added here because main branch was merged here)

* Migrate all contract functionality from mech contract on Trader to IEKit (deleted contract in https://github.com/valory-xyz/trader/pull/238). Addressess issue https://github.com/valory-xyz/IEKit/issues/150.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/skills/mech_interact_abci/behaviours/request.py [hunk 1]
```diff
@@ -294,7 +305,13 @@ def async_act(self) -> Generator:
...
             if not self._mech_requests:
                 payload = MechRequestPayload(
-                    self.context.agent_address, None, None, GNOSIS_CHAIN_ID, None, None
+                    self.context.agent_address,
+                    self.matching_round.auto_round_id(),
+                    None,
+                    None,
+                    GNOSIS_CHAIN_ID,
+                    None,
+                    None,
```
Comment (Adamantios) [inline packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: The payload does not accept `None`.

File: packages/valory/skills/mech_interact_abci/behaviours/request.py [hunk 2]
```diff
@@ -310,6 +327,7 @@ def async_act(self) -> Generator:
                 )
```
Comment (Adamantios) [inline packages/valory/skills/mech_interact_abci/behaviours/request.py:327]: This logging should not be present or at least not at an `info` level. Moreover, `serialized_data` is a generator.

File: packages/valory/skills/mech_interact_abci/behaviours/request.py (hunk 1 above)
Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: Fixed.

File: packages/valory/skills/mech_interact_abci/behaviours/request.py (hunk 2 above)
Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:327]: Please make a concrete suggestion how to fix this.

File: packages/valory/skills/mech_interact_abci/behaviours/response.py [hunk 3]
```diff
@@ -116,6 +116,8 @@ def setup(self) -> None:
 
     def set_mech_response_specs(self, request_id: int) -> None:
         """Set the mech's response specs."""
+        self.context.logger.info("set_mech_response_specs 1")
+
```
Comment (dvilelaf) [inline packages/valory/skills/mech_interact_abci/behaviours/response.py:120]: Is this maybe for debugging prurposes?

File: packages/valory/skills/mech_interact_abci/behaviours/request.py (hunk 2 above)
Comment (Adamantios) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:327]: Opened #153.

File: packages/valory/skills/mech_interact_abci/behaviours/request.py (hunk 1 above)
Comment (Adamantios) [inline packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: This won't trigger the `none_event`.

File: packages/valory/skills/mech_interact_abci/models.py [hunk 4]
```diff
@@ -65,6 +65,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.mech_contract_address: str = self._ensure(
             "mech_contract_address", kwargs, str
         )
+        self.request_price: Optional[int] = kwargs.get("request_price", None)
```
Comment (Adamantios) [inline packages/valory/skills/mech_interact_abci/models.py:68]: Why not `self._ensure(...`?

File: packages/valory/skills/mech_interact_abci/behaviours/response.py (hunk 3 above)
Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/response.py]: Fixed

File: packages/valory/skills/mech_interact_abci/models.py (hunk 4 above)
Comment (dvilelaf) [reply packages/valory/skills/mech_interact_abci/models.py:68]: Maybe this is used in more than one skill and ensure pops.

Comment (Adamantios) [reply packages/valory/skills/mech_interact_abci/models.py:68]: Can you point out which skill?

Comment (dvilelaf) [reply packages/valory/skills/mech_interact_abci/models.py:68]: That was a guess, not an actual assertion.

Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/models.py:68]: It was decided that request_price be an optional parameter. Therefore, cannot use _ensure.

File: packages/valory/skills/mech_interact_abci/behaviours/request.py (hunk 1 above)
Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: This was the case in the previous state of the repository (GNOSIS_CHAIN_ID was also present). Please open an issue to fix it in another PR.

Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: Additionally, no NONE event is specified in the FSM app:
https://github.com/valory-xyz/IEKit/blob/main/packages/valory/skills/mech_interact_abci/rounds.py#L51-L56

Comment (jmoreira-valory) [reply packages/valory/skills/mech_interact_abci/behaviours/request.py:314]: Added issue https://github.com/valory-xyz/IEKit/issues/155. This has to be addressed on a different PR, as this one is already quite complex.

**Review by dvilelaf — APPROVED** [review]

File: packages/valory/skills/mech_interact_abci/models.py (hunk 4 above)
Comment (0xArdi) [reply packages/valory/skills/mech_interact_abci/models.py:68]: This is an optional param.

**Pull request merged**